File: design-patterns/src/main/java/org/prateek/LLDProject/RideSharingApp/V2/Bike.java
package org.Prathamesh.LLDProject.RideSharingApp.V2;

public class Bike extends Vehicle {
    public Bike(String numberPlate) {
        super(numberPlate);
    }

    @Override
    public double getFarePerKm() {
        return 10;
    }
}

File: cpp/LLDProject/RideSharingApp/V2/Bike.cpp
#ifndef BIKE_CPP
#define BIKE_CPP

#include "Vehicle.cpp"

class Bike : public Vehicle {
public:
    Bike(const std::string& numberPlate) : Vehicle(numberPlate) {}
    
    double getFarePerKm() {
        return 10.0;
    }
};

#endif 